src/wp-includes/media.php

```diff
@@ -1593,6 +1606,42 @@ function wp_image_add_srcset_and_sizes( $image, $image_meta, $attachment_id ) {
 	return $image;
 }
 
+/**
+ * Filters 'img' and 'iframe' elements in post content to add 'loading' attributes.
+ *
+ * @since 5.4.0
+ *
+ * @param string $content The raw post content to be filtered.
+ * @return string Converted content with 'loading' attributes added to images.
+ */
+function wp_lazy_load_content_media( $content ) {
+	$tags = array( 'img', 'iframe' );
+
+	/**
+	 * Filters whether media in post content should be lazy-loaded with 'loading' attributes.
+	 *
+	 * @since 5.4.0
+	 *
+	 * @param bool   $enabled Whether to lazy-load content media. Default true.
+	 * @param string $content The raw post content to be filtered.
+	 */
+	if ( ! apply_filters( 'wp_lazy_load_content_media', true, $content ) ) {
+		return $content;
+	}
+
+	return preg_replace_callback(
+		'/<(' . implode( '|', $tags ) . ') [^>]+>/',
+		function( array $matches ) {
+			if ( false === strpos( $matches[0], ' loading=' ) ) {
+				$tag = $matches[1];
+				return str_replace( '<' . $tag . ' ', '<' . $tag . ' loading="lazy" ', $matches[0] );
```

Comment:
See note above about the required space after `$tag` being potentially problematic. I think this can be done better with:

```suggestion
				return preg_replace( "/(?<=<{$tag}\s)/", 'loading="lazy" ', $matches[0] );
```